src/plugins/data_explorer/public/components/sidebar/index.tsx

```diff
@@ -59,23 +59,31 @@ export const Sidebar: FC = ({ children }) => {
     }
   }, [indexPatternId, activeDataSources, dataSourceOptionList]);
 
+  const redirectToLogExplorer = useCallback(
+    (dsName: string, dsType: string) => {
+      return application.navigateToUrl(
+        `../observability-logs#/explorer?datasourceName=${dsName}&datasourceType=${dsType}`
```

Comment:
Does this assume that the plugin handling `observability-logs` is always installed?

Reply:
The s3 datasource with this redirection is for now only be able to be registered through observability plugin. Therefore it can only see s3 datasources when observability (log explorer as part of it) presented.

Reply:
What happens if I have an S3 DS configured and then connect to the cluster using an OSD that doesn't have the observability plugin? On the surface, I think it will show the DS and attempt to redirect. This should check if the plugin is available and if not, fail to redirect, and also disable selecting the DS.

Reply:
yes, currently, there is no way to register s3 datasources outside of Observability plugin. Soon in next release probably, discover will be the only data exploration interface for s3 datasources where this temporary redirection will be removed. I think it should also make sense to users that whatever datasource showed up in discover datasource selector is supported and available through using discover to avoid confusion.

Reply:
Let's make sure we have an issue to track this @mengweieric